### ISSUE TITLE
Ritsu-Cogs application

### ISSUE THREAD

**Issue by ritsu** [Twentysix26/Red-Docs#39]
List: beta

Repository name: <a href="https://github.com/ritsu/RitsuCogs">RitsuCogs</a>

Author: Ritsu / kagami#6142

Description: Some /usr/bin inspired cogs. Includes several text manipulation utilities ported from *nix.  Can read input from user message, chat log, or URL. Can chain commands together via pipes and redirect output to pastebin. Also includes an anime RSS alert cog specifically built for TokyoTosho.

**Comment by Redjumpman** [comment]
This is a great example of submitting a repo. You did well at documenting your readme, and this is the first time I don't have to ask why someone imported discord. I think the only thing here is that your TokyoTosho cog needs to be cleaned up a bit. If you run it through a linter it will find all the stuff I'm talking about.  Personally I would remove `from cogs.utils.chat_formatting import *` but it's your cog and it doesn't really matter if you clutter the namespace when it's only 600ish lines of code.

 +1 from me.

**Comment by ritsu** [comment]
Thanks for the comment! Cleaned up the code and import.

**Comment by Twentysix26** [comment]
First of all, wow, that's some neat code.  
I tested your sysinfo cog on Windows (I assume it's meant to be cross platform since it's using psutil?). It errors out at [line 50](https://github.com/ritsu/RitsuCogs/blob/80410bc8020b0f263cd8eaeb7c914e42acd18abf/sysinfo/sysinfo.py#L50) with `ValueError: commonpath() arg is an empty sequence`. Works fine on Ubuntu.

Classes names should be camelcase. That will be the name of the cog and the user expects that when using `[p]help`

I also have to take a better look at the GNU cog (from a security POV). I see that things like setting the API key are not owner restricted

**Comment by ritsu** [comment]
Thanks for the comment! 

I renamed sysinfo to SysInfo. Not sure about GNU, since it's an acronym. I guess I'm following the abbreviations note in <a href="https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles">CapitalizedWords in PEP 8</a> for GNU.

The `ValueError` seems to come from an empty list of file handles. The method that populates the list, `psutil.Process().open_files()`, seems to have limitations on Windows according to the <a href="https://pythonhosted.org/psutil/">psutil docs</a>. That may have been the cause, or else there were simply no file handles open for the bot. In any case, I added a check for empty list of file handles.

I changed the permission for pastebin api key from admin to owner. Thanks for catching that.

I try to put all user generated output for GNU in ``` comment blocks while escaping out any occurrences of ```. I may have overlooked some security issues though, so I definitely appreciate any help in finding those.
